### PR TITLE
getting_started.rst: fix list of spack deps

### PR DESCRIPTION
**Pull request by haampie**
remove curl and python-* packages, which are not necessary, and apparently don't exist on all debian/ubuntu versions

make the ubuntu list minimal in what gets installed

### CODE REVIEW

File: lib/spack/docs/getting_started.rst [hunk 1]
```diff
@@ -35,22 +35,22 @@ A build matrix showing which packages are working on which systems is shown belo
...
-         dnf install curl findutils gcc-gfortran gnupg2 hostname iproute redhat-lsb-core python3 python3-pip python3-setuptools unzip python3-boto3
+         dnf install gcc-gfortran gnupg2 redhat-lsb-core python3 unzip
 
    .. tab-item:: macOS Brew
 
       .. code-block:: console
 
          brew update
-         brew install curl gcc git gnupg zip
+         brew install gcc git gnupg zip
```
Comment (alalazo) [inline lib/spack/docs/getting_started.rst]:
```suggestion
         brew install gcc git zip
```

File: lib/spack/docs/getting_started.rst [hunk 2]
```diff
@@ -35,22 +35,22 @@ A build matrix showing which packages are working on which systems is shown belo
       .. code-block:: console
 
          apt update
-         apt install build-essential ca-certificates coreutils curl environment-modules gfortran git gpg lsb-release python3 python3-distutils python3-venv unzip zip
+         apt install bzip2 ca-certificates file g++ gcc gfortran git gnupg gzip lsb-release patch python3 tar unzip xz-utils zstd
```
Comment (alalazo) [inline lib/spack/docs/getting_started.rst]:
```suggestion
         apt install bzip2 ca-certificates file g++ gcc gfortran git gzip lsb-release patch python3 tar unzip xz-utils zstd
```

File: lib/spack/docs/getting_started.rst [hunk 3]
```diff
@@ -35,22 +35,22 @@ A build matrix showing which packages are working on which systems is shown belo
...
 
    .. tab-item:: RHEL
 
       .. code-block:: console
 
          dnf install epel-release
          dnf group install "Development Tools"
-         dnf install curl findutils gcc-gfortran gnupg2 hostname iproute redhat-lsb-core python3 python3-pip python3-setuptools unzip python3-boto3
+         dnf install gcc-gfortran gnupg2 redhat-lsb-core python3 unzip
```
Comment (alalazo) [inline lib/spack/docs/getting_started.rst]:
```suggestion
         dnf install gcc-gfortran redhat-lsb-core python3 unzip
```

**Review by alalazo — COMMENTED** [review]
LGTM, but in theory we should be able to bootstrap GnuPG

File: lib/spack/docs/getting_started.rst [hunk 4]
```diff
@@ -35,22 +35,22 @@ A build matrix showing which packages are working on which systems is shown belo
...
 
          apt update
-         apt install build-essential ca-certificates coreutils curl environment-modules gfortran git gpg lsb-release python3 python3-distutils python3-venv unzip zip
+         apt install bzip2 ca-certificates file g++ gcc gfortran git gnupg gzip lsb-release patch python3 tar unzip xz-utils zstd
 
    .. tab-item:: RHEL
 
       .. code-block:: console
 
          dnf install epel-release
```
Comment (wdconinc) [inline lib/spack/docs/getting_started.rst:44]: While this is being evaluated for minimalism: Is EPEL still needed? Originally this was for python3, right?

Comment (haampie) [reply lib/spack/docs/getting_started.rst:44]: not a rhel user, so idk

**Review by alalazo — APPROVED** [review]

File: lib/spack/docs/getting_started.rst (hunk 4 above)
Comment (wdconinc) [reply lib/spack/docs/getting_started.rst:44]: Neither am I :-)

**Pull request merged**